File: publish_docker_image.py
#!/usr/bin/env python
# -*- encoding: utf-8

import os
import subprocess


if __name__ == "__main__":
    image_name = os.environ["DOCKER_IMAGE_NAME"]
    old_version = int(os.environ["DOCKER_IMAGE_VERSION"])
    new_version = image_version + 1

    existing_makefile = open("Makefile").read()

    image_tag = image_name + ":" + str(new_version)
    subprocess.check_call(["docker", "build", "--tag", image_tag, "."])

    subprocess.check_call(["docker", "push", image_tag])

    new_makefile = existing_makefile.replace(
        "DOCKER_IMAGE_VERSION = %d" % old_version,
        "DOCKER_IMAGE_VERSION = %d" % new_version
    )
    open("Makefile", "w").write(new_makefile)

File: scripts/publish_docker_image.rb
#!/usr/bin/env ruby

image_name = ENV.fetch('DOCKER_IMAGE_NAME', nil)
old_version = ENV['DOCKER_IMAGE_VERSION'].to_i
new_version = old_version + 1

existing_makefile = File.read('Makefile')

new_image_tag = "#{image_name}:#{new_version}"

system("docker buildx build --push \
  --platform linux/amd64,linux/arm64 \
  --tag #{new_image_tag} .")

new_makefile = existing_makefile.sub(
  "DOCKER_IMAGE_VERSION = #{old_version}",
  "DOCKER_IMAGE_VERSION = #{new_version}"
)

File.open('Makefile', 'w') do |f|
  f.puts new_makefile
end
